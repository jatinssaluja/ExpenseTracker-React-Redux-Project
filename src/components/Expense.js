import React from 'react';
import {removeExpense} from '../actions/expenses';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const Expense = (props)=>(

      <Link className="list-item" to={`/edit/${props.expense.id}`}>

          <div>
             <h3 className="list-item__title">{props.expense.description}</h3>
             <span className="list-item__subtitle">{moment(props.expense.createdAt).format('MMMM Do, YYYY')}</span>
           </div>

        <h3>{numeral(props.expense.amount/100).format('$0,0.00')}</h3>

      </Link>
	);




export default Expense;
