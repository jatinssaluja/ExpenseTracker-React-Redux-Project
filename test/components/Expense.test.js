import {shallow} from 'enzyme';
import React from 'react';
import Expense from '../../src/components/Expense';
import toJSON from 'enzyme-to-json';
import expenses from '../fixtures/expenses';


test('should render Expense Component with fixture data', ()=>{

const wrapper = shallow(<Expense expense={expenses[0]}/>);
expect(toJSON(wrapper)).toMatchSnapshot();


});