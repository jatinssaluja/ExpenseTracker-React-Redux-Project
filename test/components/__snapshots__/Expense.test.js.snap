// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense Component with fixture data 1`] = `
<div>
  <p>
    Gum
  </p>
  <p>
    $1.95
     - 
    December 31st, 1969
  </p>
  <Link
    replace={false}
    to="/edit/1"
  >
    Edit
  </Link>
</div>
`;
